modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -581,13 +585,187 @@ private void initUI() {
         }
     }
 
-    private void doColumnAutoSize(TableColumnBase<?,?> column, int cellsToMeasure) {
-        double prefWidth = column.getPrefWidth();
+    private void doColumnAutoSize(int cellsToMeasure) {
+        double prefWidth = getTableColumn().getPrefWidth();
 
         // if the prefWidth has been set, we do _not_ autosize columns
         if (prefWidth == DEFAULT_COLUMN_WIDTH) {
-            TableSkinUtils.resizeColumnToFitContent(getTableSkin(), column, cellsToMeasure);
-//            getTableViewSkin().resizeColumnToFitContent(column, cellsToMeasure);
+            resizeColumnToFitContent(cellsToMeasure);
+        }
+    }
+
+    /**
+     * Resizes this {@code TableColumnHeader}'s column based on content of header and content of cells. This
+     * implementation measures the preferred width of the header, the preferred width of the first {@code maxRow} cells
+     * and sizes the column to the maximum width of all measured values. Subclass can either call this method or
+     * override it (no need to call {@code super()}) to provide their custom implementation (exclude headers, exclude
+     * null content, use min).
+     *
+     * @param maxRows the number of rows considered when resizing. If -1 is given, all rows are considered.
+     * @since 14
```

Comment:
Here's my suggestion (mind the max character length per line):

```
Resizes this {@code TableColumnHeader}'s column to fit the width of its content. The resulting column width is the maximum of the preferred width of the header cell and the preferred width of the first {@code maxRow} cells.
<p>
Subclasses can either use this method or override it (without the need to call {@code super()}) to provide their custom implementation (such as ones that exclude the header, exclude {@code null} content, compute the minimum width etc.).
```

Reply:
Seems good but I have a minor doubt on "header cell". Can the header be considered as a "cell"?

Reply:
I don't mind either way, but I think that you're right, the header is not technically a cell.

Reply:
> 
> 
> Allright @kleopatra , I have indeed removed the TableColumn argument. It is clearer now that we are resizing the TableColumn of the header.
> 

good, IMO :) Are there any tests guaranteeing that new behaviour is the same as old behaviour?

Reply:
I wouldn't know if there are enough tests on that part. I know I added one.